presto-hive/src/test/java/com/facebook/presto/hive/TestOrcFileWriter.java

```diff
@@ -0,0 +1,145 @@
+
+public class TestOrcFileWriter
+{
+    @Test(expectedExceptions = PrestoException.class, expectedExceptionsMessageRegExp = "Dummy PrestoException from mocked data sink instance")
+    public void testPrestoExceptionPropagation()
+    {
+        // This test is to verify that a PrestoException thrown by the underlying data sink implementation is propagated as is
+        OrcFileWriter orcFileWriter = new OrcFileWriter(
+                new MockDataSink(true),
+                () -> null,
+                ORC,
+                ImmutableList.of("test1"),
+                ImmutableList.of(VARCHAR),
+                NONE,
+                OrcWriterOptions.builder()
+                        .withFlushPolicy(DefaultOrcWriterFlushPolicy.builder()
+                                .withStripeMinSize(new DataSize(0, MEGABYTE))
+                                .withStripeMaxSize(new DataSize(32, MEGABYTE))
+                                .withStripeMaxRowCount(10)
+                                .build())
+                        .withRowGroupMaxRowCount(10_000)
+                        .withDictionaryMaxMemory(new DataSize(32, MEGABYTE))
+                        .build(),
+                new int[0],
+                ImmutableMap.of(),
+                HIVE_STORAGE_TIME_ZONE,
+                Optional.empty(),
+                null,
+                NOOP_WRITER_STATS,
+                NO_ENCRYPTION,
+                Optional.empty());
+
+        // Throws PrestoException with STORAGE_ERROR error code
+        orcFileWriter.commit();
+    }
+
+    @Test(expectedExceptions = PrestoException.class, expectedExceptionsMessageRegExp = "Error committing write to Hive.*")
+    public void testIOExceptionPropagation()
+    {
+        // This test is to verify that a IOException thrown by the underlying data sink implementation is converted to PrestoException.
+        OrcFileWriter orcFileWriter = new OrcFileWriter(
+                new MockDataSink(false),
+                () -> null,
+                ORC,
+                ImmutableList.of("test1"),
+                ImmutableList.of(VARCHAR),
+                NONE,
+                OrcWriterOptions.builder()
+                        .withFlushPolicy(DefaultOrcWriterFlushPolicy.builder()
+                                .withStripeMinSize(new DataSize(0, MEGABYTE))
+                                .withStripeMaxSize(new DataSize(32, MEGABYTE))
+                                .withStripeMaxRowCount(10)
+                                .build())
+                        .withRowGroupMaxRowCount(10_000)
+                        .withDictionaryMaxMemory(new DataSize(32, MEGABYTE))
+                        .build(),
+                new int[0],
+                ImmutableMap.of(),
+                HIVE_STORAGE_TIME_ZONE,
+                Optional.empty(),
+                null,
+                NOOP_WRITER_STATS,
+                NO_ENCRYPTION,
+                Optional.empty());
```

Comment:
nit: instead of duplicating just have a method called createOrcFileWriter which takes in a data sink (or boolean) which you can flip between true and false to trigger the different paths.